dist/images/daemonset.sh

```diff
@@ -275,6 +289,15 @@ ovn_image=${image} \
   ovn_sb_raft_port=${ovn_sb_raft_port} \
   j2 ../templates/ovnkube-db-raft.yaml.j2 -o ../yaml/ovnkube-db-raft.yaml
 
+if [[ ! -z "${ovn_webhook_ca_pem_b64}" ]]; then 
+  ovn_webhook_image=${ovn_webhook_image} \
+    ovn_image_pull_policy=${image_pull_policy} \
+    ca_pem_64=${ovn_webhook_ca_pem_b64} \
+    j2 ../templates/ovnkube-webhook.yaml.j2 -o ../yaml/ovnkube-webhook.yaml
+else 
+  echo "Error: cannot deploy ovn-kubernetes webhook component without a caBundle value. Please generate a key, base64 encrypt it and pass it to this script using: \"--webhook-ca-pem-b64\""
```

Comment:
If the webhook uses a system-generated certificate then you can omit the `caBundle` line in the webhook object. I think it would be better to just change the webhook template to do that (if `ca_pem_64` is empty)

Reply:
I don't have that much experience with certificates, so this is going to sound like a silly question: what is a "system-generated certificate"? Is it something generated by a cluster component (ex: the proxy controller / pki controller in the OpenShift CNO), which are generating cluster-wide certs, as supposed to manually using `openssl`?

Reply:
No, the cluster has a CA of its own which is used for, eg, the apiserver certificate, and the kubelet certificates on each node.

For reasons I don't entirely remember, CNO can't make use of that infrastructure at install time before the cluster network is actually running, so we generate our own certs for everything we use. But in theory, in a non-OCP cluster, they might be able to have the ovn-kube webhook cert be signed by the cluster CA, in which case you wouldn't need to include the CA in the webhook object, because the apiserver would already know it.